src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -35,6 +35,8 @@ namespace OpenTelemetry.Instrumentation.AspNetCore.Implementation
     internal class HttpInListener : ListenerHandler
     {
         internal const string ActivityOperationName = "Microsoft.AspNetCore.Hosting.HttpRequestIn";
+        internal static readonly bool IsNet7OrGreater = typeof(HttpRequest).Assembly.GetName().Version.Major >= 7;
```

Comment:
consider putting this inside try..catch.

Reply:
Added